api/go/config/appconfig.pb.go

```diff
@@ -223,6 +223,9 @@ type AppInstanceConfig struct {
 	// will override default capabilities with configuration
 	// defined in org.mobyproject.config label of image provided by linuxkit
 	Service bool `protobuf:"varint,20,opt,name=service,proto3" json:"service,omitempty"`
+	// All changes to the cloud-init config are tracked using this version field -
+	// once the version is changed cloud-init tool restarts in a guest.
+	CloudInitVersion uint32 `protobuf:"varint,21,opt,name=cloudInitVersion,proto3" json:"cloudInitVersion,omitempty"`
```

Comment:
Having explicit field for user-data version is a good idea as this way we can control the behavior of cloud-init script.

But as I can see we want to detach version field from [UUIDandVersion](https://github.com/lf-edge/eve/blob/847481f701ea7bc904a6187ddebe6620f5e36714/api/proto/config/devcommon.proto#L14) message, which has string type. Would it be reasonable to have string here as well? My understanding that in general version may keep not only one number but something like `v1.1` or another useful information. We can check if this field filled with information or not instead of compare with 0.

Reply:
Oh, I see, i.e. it's closer to the [count](https://github.com/lf-edge/eve/blob/847481f701ea7bc904a6187ddebe6620f5e36714/api/proto/config/appconfig.proto#L17) to request operations, and allow us to simplify controllers's logic. Make sense, thank you for details.